src/pages/MatchExperiencesCatalogScreen/MatchExperiencesCatalogScreen.tsx

```diff
@@ -1,7 +1,157 @@
-// import classes from './matchExperiences-catalog-screen.module.scss';
+import { ShareMatchExperienceModal } from '@components/ShareMatchExperienceModal';
+import classes from './match-experiences-catalog-screen.module.scss';
+import { PlusOutlined } from '@ant-design/icons';
+import { FaFutbol } from 'react-icons/fa';
+import { Button, Select, Spin } from 'antd';
 
-export interface MatchExperiencesCatalogScreenProps {}
+import { useState } from 'react';
+import { MatchExperienceService } from '@/api/services/match-experience.service';
+import { MatchDetails } from '@components/MatchDetails';
+import { Screen } from '@components/Screen';
+import { useQuery } from '@tanstack/react-query';
+import { getPictureFullUrl } from '@/utils/picture.utils.ts';
+import moment from 'moment';
+import { Heart, MessageCircle } from 'lucide-react';
+import { ROUTES } from '@/constants/routes.const';
+import { useNavigate } from 'react-router-dom'; // ✅ Import for navigation
 
-export const MatchExperiencesCatalogScreen = () => {
-    return <div></div>;
+export interface MatchExperiencesCatalogScreenProps {
+    mode: 'all' | 'my';
+}
+
+const currentUserId = '67d592743f1f3c317f54c7de'; // creator
+const PageItemsLimit = 5; // Number of items per page
+
+export const MatchExperiencesCatalogScreen = ({ mode }: MatchExperiencesCatalogScreenProps) => {
+    const [page, setPage] = useState(1);
+    const [sortBy, setSortBy] = useState('date');
+
+    const navigate = useNavigate();
+
+    const {
+        data: matchExperiencesData,
+        error: matchExperiencesError,
+        isPending: matchExperiencesIsPending,
+        refetch,
+    } = useQuery({
+        queryKey:
+            mode === 'all' ? ['matchExperiences', page, sortBy] : ['userMatchExperiences', page, sortBy, currentUserId],
+        queryFn: () =>
+            mode === 'all'
+                ? MatchExperienceService.getAllMatchExperience(page, PageItemsLimit, sortBy)
+                : MatchExperienceService.getAllMatchExperiencesByUserId(currentUserId, page, PageItemsLimit, sortBy),
+    });
+
+    const totalPages = matchExperiencesData?.totalPages || 1;
+    const matchExperiences = matchExperiencesData?.experiences || [];
+
+    const [isModalOpen, setIsModalOpen] = useState(false);
+
+    if (matchExperiencesError) {
+        console.error('Error', matchExperiencesError);
+        return <div>Error has occurred</div>;
+    }
+
+    return (
+        <Screen className={classes.container}>
+            <div className={classes.content}>
+                <div className={classes.header}>
+                    <h1>Home Screen</h1>
```

Comment:
```suggestion
           <h1>{mode === 'all'  ?  "Home Screen" :  "My Experiences"</h1>  
```
Currently shows same title for both pages

Reply:
07e831978f911ce672e2a285203e2bcea22aadf8